core/backend/src/IModelDb.ts

```diff
@@ -2809,6 +2809,10 @@ export class BriefcaseDb extends IModelDb {
       // Restart default txn to trigger events when watch file is changed by some other process.
       const watcher = fs.watch(briefcaseDb.watchFilePathName, { persistent: false }, () => {
         nativeDb.restartDefaultTxn();
+        const changeset = briefcaseDb.nativeDb.getCurrentChangeset();
+        if (briefcaseDb.changeset.id !== changeset.id || briefcaseDb.changeset.index !== changeset.index) {
```

Comment:
@pmconne can you also call `touch.sync(this.watchFilePathName)` in `pushChanges` (and possibly `pullChanges`)? otherwise, the watcher wont be notified. 

Sorry for not mentioning it in the issue, but the unit tests manually "touch the watch file", following the existing `watchForChanges` test

Reply:
Yeah, totally forgot to add that.